microfs.py

```diff
@@ -56,10 +56,16 @@ def raw_on(serial):
     """
     Puts the device into raw mode.
     """
-    serial.write(b'\x03')  # Send CTRL-C to break out of loop.
-    serial.read_until(b'\n>')  # Flush buffer until prompt.
-    serial.write(b'\x01')  # Go into raw mode.
-    serial.read_until(b'\r\n>OK')  # Flush buffer until raw mode prompt.
+    attempts_left = 3
+    bytes_read = b''
+    while not bytes_read.endswith(b'\n>>>') and attempts_left:
+        serial.write(b'\x03')    # Send CTRL-C to break out of loop.
+        bytes_read = serial.read_until(b'\n>>>')  # Flush buffer until prompt.
+        attempts_left -= 1
+    if not attempts_left:
```

Comment:
This condition will raise the error even if the last attempt was successful, better change to a `while...else` for clarity.